packages/grid/_modules_/grid/components/containers/GridRootStyles.ts

```diff
@@ -212,6 +212,9 @@ export const GridRootStyles = styled('div', {
           },
         },
       },
+      [`&:last-child:not(.${gridClasses['row--lastBeforeEmpty']}) .${gridClasses.cell}`]: {
```

Comment:
This selector targets the last visible row that the virtualization decided to render. `:last-child` doesn't mean the last row of the dataset. Also, could we simplify it? Maybe target a row which has a `.MuiDataGrid-row--last` (being the true last row)?

Scrolling https://deploy-preview-3519--material-ui-x.netlify.app/components/data-grid/#commercial-version slowly I can see that sometimes some cells don't have the border bottom.

![image](https://user-images.githubusercontent.com/42154031/150851092-a284c15d-d7d1-4c41-9f16-0f9858f318a5.png)

This is much more prominent depending on the value of `rowThreshold`: https://codesandbox.io/s/datagridprodemo-material-demo-forked-jsnbs?file=/demo.js

Reply:
Updated. I added the `row-last` class.